VideoLocker/src/main/java/org/edx/mobile/view/CourseCombinedInfoFragment.java

```diff
@@ -345,23 +340,71 @@ private void populateFacePile(){
 
     }
 
-    private View generateAnnouncementView(AnnouncementsModel model){
+    private void populateAnnouncements(List<AnnouncementsModel> announcementsList) {
+        if(announcementsList !=null && announcementsList.size()>0) {
+            hideEmptyAnnouncementMessage();
 
-        View convertView = inflater.inflate(R.layout.row_announcement_list, null);
+            // stop existing views before we remove them from the layout
+            for (AnnouncementView view : announcementViewList) {
+                view.stop();
+            }
+            announcementViewList.clear();
+            announcementContainer.removeAllViews();
+
+            for (AnnouncementsModel m : announcementsList) {
+                AnnouncementView announcementView = generateAnnouncementView(m);
+                announcementViewList.add(announcementView);
+
+                announcementContainer.addView(announcementView.getView());
```

Comment:
I ended up going with one long web view on iOS. It's a little funky because the content isn't really formatted for mobile as I mentioned below. Also, it makes it easy for course content to infect that part of the app UI. So it's a better solution, but I'm not sure it's right long term.

Is it possible to use ListView since the height of the webview content changes as it loads?